jupyter/d2l-java/chapter_convolutional-modern/batch-norm.ipynb

```diff
@@ -0,0 +1,1020 @@
+    "%mavenRepo snapshots https://oss.sonatype.org/content/repositories/snapshots/\n",
+    "\n",
+    "%maven ai.djl:api:0.7.0-SNAPSHOT\n",
+    "%maven ai.djl:model-zoo:0.7.0-SNAPSHOT\n",
+    "%maven ai.djl:basicdataset:0.7.0-SNAPSHOT\n",
+    "%maven org.slf4j:slf4j-api:1.7.26\n",
+    "%maven org.slf4j:slf4j-simple:1.7.26\n",
+    "    \n",
+    "%maven ai.djl.mxnet:mxnet-engine:0.7.0-SNAPSHOT\n",
+    "%maven ai.djl.mxnet:mxnet-native-auto:1.7.0-a"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%loadFromPOM\n",
+    "<dependency>\n",
+    "    <groupId>tech.tablesaw</groupId>\n",
+    "    <artifactId>tablesaw-jsplot</artifactId>\n",
+    "    <version>0.30.4</version>\n",
+    "</dependency>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%load ../utils/plot-utils.ipynb"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import ai.djl.Device;\n",
+    "import ai.djl.Model;\n",
+    "import ai.djl.basicdataset.FashionMnist;\n",
+    "import ai.djl.metric.Metrics;\n",
+    "import ai.djl.ndarray.NDArray;\n",
+    "import ai.djl.ndarray.NDList;\n",
+    "import ai.djl.ndarray.NDManager;\n",
+    "import ai.djl.ndarray.types.Shape;\n",
+    "import ai.djl.nn.*;\n",
+    "import ai.djl.nn.convolutional.Conv2d;\n",
+    "import ai.djl.nn.core.Linear;\n",
+    "import ai.djl.nn.pooling.Pool;\n",
+    "import ai.djl.training.DefaultTrainingConfig;\n",
+    "import ai.djl.training.EasyTrain;\n",
+    "import ai.djl.training.ParameterStore;\n",
+    "import ai.djl.training.Trainer;\n",
+    "import ai.djl.training.dataset.ArrayDataset;\n",
+    "import ai.djl.training.dataset.Dataset;\n",
+    "import ai.djl.training.evaluator.Accuracy;\n",
+    "import ai.djl.training.listener.TrainingListener;\n",
+    "import ai.djl.training.loss.Loss;\n",
+    "import ai.djl.training.optimizer.Optimizer;\n",
+    "import ai.djl.training.optimizer.learningrate.LearningRateTracker;\n",
+    "import ai.djl.util.PairList;\n",
+    "\n",
+    "import java.io.IOException;\n",
+    "import java.util.HashMap;\n",
+    "import java.util.Map;\n",
+    "\n",
+    "import tech.tablesaw.api.*;\n",
+    "import tech.tablesaw.plotly.api.*;\n",
+    "import tech.tablesaw.plotly.components.*;\n",
+    "import tech.tablesaw.plotly.Plot;\n",
+    "import tech.tablesaw.plotly.components.Figure;\n",
+    "import org.apache.commons.lang3.ArrayUtils;"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "attributes": {
+     "classes": [],
+     "id": "",
+     "n": "72"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "// utility class to store Batch Norm parameters such as Y, moving mean & moving Var.\n",
+    "class BatchNormObj {\n",
+    "\n",
+    "    private NDArray Y;\n",
```

Comment:
Why we are keeping Y here, what did y do?

Reply:
It's been used in the forward function (for returning Y) and also in the batchNormUpdate method. We need the other 2 as they are constantly changing too.

Reply:
@lanking520 is this okay for you or I need to change anything with this ?